mso/resource_mso_schema_site_bd.go

```diff
@@ -279,14 +280,31 @@ func resourceMSOSchemaSiteBdUpdate(d *schema.ResourceData, m interface{}) error
 	bdRefMap["templateName"] = bd_template_name
 	bdRefMap["bdName"] = bdName
 
-	path := fmt.Sprintf("/sites/%s-%s/bds/%s", siteId, templateName, bdName)
-	bdStruct := models.NewSchemaSiteBd("replace", path, mac, bdRefMap, host)
+	payloadCon := container.New()
+	payloadCon.Array()
+
+	err := setPatchPayloadToContainer(payloadCon, "replace", fmt.Sprintf("/sites/%s-%s/bds/%s/bdRef", siteId, templateName, bdName), bdRefMap)
+	if err != nil {
+		return err
+	}
+
+	err = setPatchPayloadToContainer(payloadCon, "replace", fmt.Sprintf("/sites/%s-%s/bds/%s/hostBasedRouting", siteId, templateName, bdName), host)
+	if err != nil {
+		return err
+	}
```

Comment:
do we need a condition before this? will it error out if there is no host?

Reply:
The variable is initialised here so set to false so will always have a host_route: https://github.com/CiscoDevNet/terraform-provider-mso/pull/276/files#diff-86bef212dcb5a9532691f6ae8c3f0365cff9505f20b6642a9dcc55836c7a6da5R263.